backend/connector-integration/src/connectors/cybersource/transformers.rs

```diff
@@ -0,0 +1,4440 @@
+use base64::Engine;
+use jsonwebtoken as jwt;
+use serde_json::Value;
+
+use common_utils::{
+    consts::{NO_ERROR_CODE, NO_ERROR_MESSAGE},
+    ext_traits::OptionExt,
+    pii,
+    types::{SemanticVersion, StringMajorUnit},
+};
+
+use crate::unimplemented_payment_method;
+use crate::{connectors::cybersource::CybersourceRouterData, types::ResponseRouterData, utils};
+use cards;
+use domain_types::{
+    connector_flow::{
+        Authenticate, Authorize, Capture, PostAuthenticate, PreAuthenticate, SetupMandate, Void,
+    },
+    connector_types::{
+        MandateReference, PaymentFlowData, PaymentVoidData, PaymentsAuthenticateData,
+        PaymentsAuthorizeData, PaymentsCaptureData, PaymentsPostAuthenticateData,
+        PaymentsPreAuthenticateData, PaymentsResponseData, PaymentsSyncData, RefundFlowData,
+        RefundSyncData, RefundsData, RefundsResponseData, RepeatPaymentData, ResponseId,
+        SetupMandateRequestData,
+    },
+    errors::{self, ConnectorError},
+    payment_address::Address,
+    payment_method_data::{
+        self, ApplePayWalletData, GPayPredecryptData, GooglePayWalletData, NetworkTokenData,
+        PaymentMethodData, PaymentMethodDataTypes, RawCardNumber, SamsungPayWalletData, WalletData,
+    },
+    router_data::{
+        AdditionalPaymentMethodConnectorResponse, ApplePayPredecryptData, ConnectorAuthType,
+        ErrorResponse, GooglePayDecryptedData, NetworkTokenNumber, PaymentMethodToken,
+        PazeDecryptedData,
+    },
+    router_data_v2::RouterDataV2,
+    router_response_types::RedirectForm,
+    utils::CardIssuer,
+};
+use error_stack::{report, ResultExt};
+use hyperswitch_masking::{ExposeInterface, PeekInterface, Secret};
+use serde::{Deserialize, Serialize};
+pub const BASE64_ENGINE: base64::engine::GeneralPurpose = base64::engine::general_purpose::STANDARD;
+pub const REFUND_VOIDED: &str = "Refund request has been voided.";
+
+fn get_card_number_string<T: PaymentMethodDataTypes + 'static>(
+    card_number: &RawCardNumber<T>,
+) -> String {
+    use std::any::{Any, TypeId};
+
+    let any_ref: &dyn Any = card_number;
+
+    match TypeId::of::<T>() {
+        id if id == TypeId::of::<payment_method_data::DefaultPCIHolder>() => any_ref
+            .downcast_ref::<RawCardNumber<payment_method_data::DefaultPCIHolder>>()
+            .map(|c| c.peek().to_string())
+            .unwrap_or_default(),
+        _ => any_ref
+            .downcast_ref::<RawCardNumber<payment_method_data::VaultTokenHolder>>()
+            .map(|c| c.peek().to_string())
+            .unwrap_or_default(),
+    }
+}
```

Comment:
IMO we should avoid runtime type checks. 
```
                    None => domain_types::utils::get_card_issuer(&get_card_number_string(
                        &ccard.card_number,
                    ))
  ```
  wherever this function is used, either we can add and use impl function `get_card_issuer` of Card<T> type or We can add a peek impl to RawCardNumber<T>. 
  any thoughts on this ?